### PR TITLE
Add uikit properties parsing

### PR DESCRIPTION
**Pull request by kerrmarin**
Counterpart to https://github.com/design-ops/stylable-swiftui/pull/32

### CODE REVIEW

**Comment by kerrmarin** [comment]
@atomoil ping

File: lib/processLayerStyles.js [hunk 1]
```diff
@@ -16,32 +16,53 @@ const PATTERN_FILL_TYPE = {
...
         addOptionalProperty("border", data.style, layerBorder(layerStyle) )
-        addOptionalProperty("shadow", data.style, layerShadow(layerStyle) )
         addOptionalProperty("blur", data.style, layerBlur(layerStyle) )
         // Apply the corner radius modifier
         addOptionalProperty("cornerRadius", data.style, cornerRadius(layerStyle, data.name, modifiers.radius) )
 
-        ret.push(data)
+        styles.push(data)
+
+        // Get properties
```
Comment (atomoil) [inline lib/processLayerStyles.js:38]: What properties?

File: lib/processLayerStyles.js [hunk 2]
```diff
@@ -16,32 +16,53 @@ const PATTERN_FILL_TYPE = {
...
+      let fill = fills[i];
+      if (fill.fillType == FILL_TYPE.SOLID_COLOR && fill.isEnabled == true) {
+          backgroundArray.push(`.backgroundColor(UIColor(red: ${fill.color.red}, green: ${fill.color.green}, blue: ${fill.color.blue}, alpha: ${fill.color.alpha}))`)
+      }
+    }
+  }
+  return backgroundArray;
+}
+
 const layerFill = (layerStyle, outputFolder, sketchFile) => {
```
Comment (atomoil) [inline lib/processLayerStyles.js:66]: If this is now handling fill and shadow then it should named appropriately

File: lib/processLayerStyles.js [hunk 3]
```diff
@@ -16,32 +16,53 @@ const PATTERN_FILL_TYPE = {
 }
 
 const processLayerStyles = async (layerStyles, filepath, outputFolder, modifiers) => {
-    let ret = []
+    let styles = []
+    let properties = []
```
Comment (atomoil) [inline lib/processLayerStyles.js:20]: This should be clearer - is styles == swiftUI and properties == UIKit ? If so, name them to be clearer

File: lib/processTextStyles.js [hunk 4]
```diff
@@ -40,12 +41,24 @@ async function processTextStyles(textStyles, outputFolder) {
...
         addOptionalProperty("strikethrough", data.text, textStrikethrough(encodedAttributes))
-        ret.push(data)
+        styles.push(data)
+
+        // Get properties
+        const propertyData = {
+          name: style.name,
+          properties: [
+            `.textColor(UIColor(red: ${color.red}, green: ${color.green}, blue: ${color.blue}, alpha: ${color.alpha}))`,
+            `.kerning(${encodedAttributes.kerning})`,
```
Comment (atomoil) [inline lib/processTextStyles.js:51]: Is kerning always present?

File: lib/processTextStyles.js [hunk 5]
```diff
@@ -40,12 +41,24 @@ async function processTextStyles(textStyles, outputFolder) {
...
+            `.font(UIFont(name: "${fontAttributes.name}", size: ${fontAttributes.size})!)` // using ! to unwrap the font. Should we do this? IMHO the font not being there is a programmer error
```
Comment (atomoil) [inline lib/processTextStyles.js:52]: This should follow the name / value pattern of the SwiftUI for consistency, unless there’s a good reason not to :-)

**Review by atomoil — CHANGES_REQUESTED** [review]

**Comment by kerrmarin** [comment]
@atomoil I've (I think) addressed all your points

File: lib/processLayerStyles.js [hunk 6]
```diff
@@ -16,33 +16,54 @@ const PATTERN_FILL_TYPE = {
...
         // Apply the corner radius modifier
         addOptionalProperty("cornerRadius", data.style, cornerRadius(layerStyle, data.name, modifiers.radius) )
 
-        ret.push(data)
+        swiftUIStyles.push(data)
+
+        // For each layer style we also pull out the UIKit properties we want to expose
+        const propertyData = {
+          name: style.name,
+          properties: layerFillProperty(layerStyle)
```
Comment (atomoil) [inline lib/processLayerStyles.js:41]:
```suggestion
          properties: layerFillUIKitBackground(layerStyle)
```
?

File: lib/processTextStyles.js [hunk 7]
```diff
@@ -40,13 +40,30 @@ async function processTextStyles(textStyles, outputFolder) {
...
         addOptionalProperty("strikethrough", data.text, textStrikethrough(encodedAttributes))
-        ret.push(data)
+        swiftUIStyles.push(data)
+
+        // Get properties we want to make accessible for UIKit
+        const propertyData = {
+          name: style.name,
+          properties: [
+            { name: "textColor", value: `UIColor(red: ${color.red}, green: ${color.green}, blue: ${color.blue}, alpha: ${color.alpha})`},
+            // using ! to unwrap the font. Should we do this? IMHO the font not being there is a programmer error
```
Comment (atomoil) [inline lib/processTextStyles.js:50]: can't you use `addOptionalProperty` ?

Comment (atomoil) [reply lib/processTextStyles.js:50]: oh, sorry, re-read and now understand the comment is for swift, not convertor

File: lib/processTextStyles.js [hunk 8]
```diff
@@ -40,13 +40,30 @@ async function processTextStyles(textStyles, outputFolder) {
...
         addOptionalProperty("underline", data.text, textUnderline(encodedAttributes))
         addOptionalProperty("strikethrough", data.text, textStrikethrough(encodedAttributes))
-        ret.push(data)
+        swiftUIStyles.push(data)
+
+        // Get properties we want to make accessible for UIKit
+        const propertyData = {
+          name: style.name,
+          properties: [
+            { name: "textColor", value: `UIColor(red: ${color.red}, green: ${color.green}, blue: ${color.blue}, alpha: ${color.alpha})`},
```
Comment (atomoil) [inline lib/processTextStyles.js:49]: can these properties move into functions, instead of being explicitly declared here?

Comment (atomoil) [inline lib/processTextStyles.js:49]:
```suggestion
            { name: "textColor", value: getUIColor(color)},
```

**Comment by atomoil** [comment]
Did you mean to remove kerning from the UIKit side?

**Comment by kerrmarin** [comment]
@atomoil updated 👍 

I didn't remove the ker(r)nin, I just moved it to being optional; so I'm checking if it's there and if it is i'll add it.

**Comment by kerrmarin** [comment]
@atomoil ping

**Review by atomoil — APPROVED** [review]

**Pull request merged**